src/test/java/org/pentaho/di/trans/dataservice/optimization/OptimizationImpactInfoTest.java

```diff
@@ -0,0 +1,42 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2016 by Pentaho : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.dataservice.optimization;
+
+import org.junit.Test;
+import org.pentaho.di.core.util.Assert;
+
+public class OptimizationImpactInfoTest {
+
+  @Test
+  public void getDescriptionForNullErrorMsg() throws Exception {
+    OptimizationImpactInfo info = new OptimizationImpactInfo( "test" );
+
+    try {
+      throw new NullPointerException();
```

Comment:
This is confusing--why throw an exception?  Why not just set some arbitrary text?  And your Assert below checks that the description is not empty, but doesn't do anything to verify that the text set earlier is present.